### PR TITLE
KOGITO-3604 - Assign constant value to sub process model

### PR DESCRIPTION
**Pull request by cristianonicolai**
Depends on https://github.com/kiegroup/kogito-runtimes/pull/880

### CODE REVIEW

**Review by tiagodolphine — APPROVED** [review]

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Comment by cristianonicolai** [comment]
@MarianMacik any feedback on this?

File: jbpm/jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/CallActivityHandler.java [hunk 1]
```diff
@@ -189,30 +192,31 @@ protected void readDataInputAssociation(org.w3c.dom.Node xmlNode, SubProcessNode
...
+				}
+				Object result = null;
+				Object from = nl.item(0);
+				if (from instanceof Text) {
+					result = ((Text) from).getTextContent();
+				} else {
+					result = nl.item(0);
+				}
+				subProcessNode.addInAssociation(
+						new DataAssociation(result.toString(), dataInputs.get(to), singletonList(new Assignment(null, result.toString(), to)), null));
```
Comment (MarianMacik) [inline jbpm/jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/CallActivityHandler.java:218]: I assume this was added just to distinguish between assigning a constant and copying the value of the parent process variable to the subprocess model.

**Review by MarianMacik — APPROVED** [review]
For some reason `ProcessContainerGenerator` and `ProcessContainerTemplate` changes are displayed among the changed files but they have been already merged by #880. So probably just a GitHub issue :) Looks good.

File: jbpm/jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/CallActivityHandler.java (hunk 1 above)
Comment (cristianonicolai) [reply jbpm/jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/CallActivityHandler.java:218]: @MarianMacik correct.

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Comment by cristianonicolai** [comment]
jenkins please retest this

**Pull request merged**